### PR TITLE
feat: Return last anchored state

### PR DESCRIPTION
**Pull request by ukstv**
3id-did-resolver returns now only the latest anchored state, or genesis state for vanilla DID URL like `did:3:foo`. See #1520

### CODE REVIEW

File: packages/3id-did-resolver/src/index.ts [hunk 1]
```diff
@@ -78,6 +79,22 @@ export function wrapDocument(content: any, did: string): DIDDocument | null {
   }, startDoc)
 }
 
+/**
+ * Return last anchor log entry, or genesis if no anchors found
+ */
+function lastAnchorOrGenesisEntry(log: LogEntry[]): LogEntry {
```
Comment (oed) [inline packages/3id-did-resolver/src/index.ts:85]: Is recursion better than a for loop here?

**Review by oed — COMMENTED** [review]
Looks good, just one question.
Also please try to provide more info in the PR description. Other ppl have minimal context here.

File: packages/3id-did-resolver/src/index.ts (hunk 1 above)
Comment (ukstv) [reply packages/3id-did-resolver/src/index.ts:85]: It is more idiomatic.

**Comment by ukstv** [comment]
@oed there is an issue linked to PR. I can copy-paste the description here, but what is the point of linking the issue then?

298px-Prap

File: packages/3id-did-resolver/src/index.ts (hunk 1 above)
Comment (ukstv) [reply packages/3id-did-resolver/src/index.ts:85]: changed to `for` loop anyway

**Review by oed — APPROVED** [review]
Approved once tests are passing. 
My bad @ukstv Didn't see the linked issue.

**Comment by oed** [comment]
But a short description like the one you added is always nice ;)

**Pull request merged**